src/wallet/wallet.cpp

```diff
@@ -1958,7 +1958,13 @@ bool CWallet::CreateTransaction(const vector<pair<CScript, int64_t> >& vecSend,
 
                         // Reserve a new key pair from key pool
                         CPubKey vchPubKey;
-                        assert(reservekey.GetReservedKey(vchPubKey)); // should never fail, as we just unlocked
+                        bool ret;
+                        ret = reservekey.GetReservedKey(vchPubKey);
+                        if (!ret)
```

Comment:
nit:
```suggestion
                        if (!reservekey.GetReservedKey(vchPubKey))
```

Reply:
While I agree with the nit we probably want to keep it as upstream as possible.

Reply:
@denravonska Please see https://github.com/bitcoin/bitcoin/blob/629c4ab2e31325614c5b5bb9636705dde4659bb3/src/wallet/spend.cpp#L658.